notes/push/sync.sh

```diff
@@ -0,0 +1,104 @@
+
+
+rm $BASEDIR/db.sqlite
+sqlite3 $BASEDIR/db.sqlite <<'EOF'
+CREATE TABLE IF NOT EXISTS cities (
+    id           INTEGER PRIMARY KEY,
+    name         TEXT
+);
+INSERT INTO cities VALUES (1, 'Vilnius');
+EOF
+sqlite3 $BASEDIR/db.sqlite "SELECT * FROM cities;"
+#| +----+---------+
+#| | id |  name   |
+#| +----+---------+
+#| | 1  | Vilnius |
+#| +----+---------+
+
+cat > $BASEDIR/manifest.txt <<EOF
+d | r | b | m | property  | type    | ref                          | source       | prepare | level | access
+$DATASET                  |         |                              |              |         |       |
+  | db                    | sql     |                              | sqlite:///$BASEDIR/db.sqlite | | |
+  |   |   | City          |         | id                           | cities       |         | 4     |
+  |   |   |   | id        | integer |                              | id           |         | 4     | open
+  |   |   |   | name      | string  |                              | name         |         | 4     | open
+EOF
+poetry run spinta copy $BASEDIR/manifest.txt -o $BASEDIR/manifest.csv
+cat $BASEDIR/manifest.csv
+poetry run spinta show
+
+# notes/spinta/server.sh    Run migrations
+
+test -n "$PID" && kill $PID
+poetry run spinta run &>> $BASEDIR/spinta.log &; PID=$!
+tail -50 $BASEDIR/spinta.log
+
+# notes/spinta/client.sh    Configure client
+
+uuidgen
+VILNIUS=646d0f3f-1d6d-493e-b1b9-b6d1472d36fb
+http POST "$SERVER/$DATASET/City" $AUTH _id=$VILNIUS id:=1 name=Vilnius
+KAUNAS=c4c53887-9332-4643-8060-d231ddf76c1d
+http POST "$SERVER/$DATASET/City" $AUTH _id=$KAUNAS id:=2 name=Kaunas
+
+http GET "$SERVER/$DATASET/City?format(ascii)"
+
+rm $BASEDIR/keymap.db
+rm $BASEDIR/push/localhost.db
+poetry run spinta push $BASEDIR/manifest.csv -o test@localhost --sync
+#| {
+#|     '_id': '85828ce4-5264-4fc0-bd9a-699ee91f6298',
+#|     '_op': 'insert',
+#|     '_type': 'push/sync/City',
+#|     'id': 1,
+#|     'name': 'Vilnius',
+#| }
+#| Server response (status=400):
+#|     {
+#|         'errors': [
+#|             {
+#|                 'code': 'UniqueConstraint',
+#|                 'context': {
+#|                     'attribute': 'id',
+#|                     'component': 'spinta.components.Property',
+#|                     'dataset': 'push/sync',
+#|                     'entity': 'cities',
+#|                     'manifest': 'default',
+#|                     'model': 'push/sync/City',
+#|                     'property': 'id',
+#|                     'resource': 'db',
+#|                     'resource.backend': 'push/sync/db',
+#|                     'schema': '5',
+#|                 },
+#|                 'message': 'Given value already exists.',
+#|                 'template': 'Given value already exists.',
+#|                 'type': 'property',
+#|             },
+#|         ],
+#|     }
+# FIXME: It looks, that sync was not done, because an attempt to create an
+#        existing City was made.
```

Comment:
If `--sync` would work correctly, this error would not happen.

Reply:
Decided that this task will be split into push state sync and keymap sync, currently only keymap part is done